test/terraform/modules/eks_cluster/main.tf

```diff
@@ -0,0 +1,39 @@
+locals {
+  iam_role_permissions_boundary = "arn:aws:iam::${var.account_id}:policy/resource-provisioner-boundary"
+}
+
+module "eks_cluster" {
+  source  = "terraform-aws-modules/eks/aws"
+  version = "20.31.4"
+
+  cluster_name    = var.name
+  cluster_version = "1.31"
+
+  cluster_endpoint_private_access = true
+  cluster_endpoint_public_access  = true
```

Comment:
Given that this cluster will run permanently, what's our strategy to protect ourself against abuse from the random internet? Not entirely sure how auth works in this case.

Reply:
Looks like [it's secured via IAM](https://docs.aws.amazon.com/eks/latest/userguide/cluster-endpoint.html), so it would be good to understand where the role requirements are configured.

Reply:
Good call out, was thinking we'd add the gha cidr blocks eventually: https://docs.aws.amazon.com/eks/latest/userguide/cluster-endpoint.html#modify-endpoint-access

Reply:
Interesting idea, maybe we can put it into a dedicated ticket to not lose track of it

Reply:
For IAM, since we're enabling `enable_cluster_creator_admin_permissions` the resource provisioner gets admin access to the cluster, we don't have access unless we add another access entry by default.

Reply:
Given that we have access to the credentials for that user, it should be enough to have that configured I guess. LGTM then